### ISSUE TITLE
NullPointerException in JpaCriteriaQueryBackend.getOrderList

### ISSUE THREAD

**Issue by markuslohn** [crnk-project/crnk-framework#558]
I setup on a Java EE 7.0 server (Oracle WebLogic) using 3.0.20190714142556 with a simple resource (customer) loaded as JPA Entity with JpaEntityRepositoryBase from a database. The application can be deployed without problems. crnk.io will be used with JAX-RS in this application. However when accessing the resource (customer) with findAll() I got a NullPointerException in `JpaCriteriaQueryBackend.getOrderList()`.

Tried to use the application in Oracle WebLogic Server 12.2.1.3 and also Payara Server. In both cases the same error occurred.

The following code line in this class returns a null instead of an empty list I guess: criteriaQuery.getOrderList().

```
@Override
  public List<Order> getOrderList() {
     return new ArrayList<>(criteriaQuery.getOrderList());

  }
```

Currently a NullPointerException was reported in line 109 of JpaCriteriaQueryBackend. Expected is that all available customers will be loaded as result.

Expected result is to get the list of customers from the database.

I attached the example implementation. The stackstrace as screenshot and a fix I tested to workaround this problem.

**Comment by markuslohn** [comment]
![crnk-stacktrace](https://user-images.githubusercontent.com/25202280/62628021-d41dd280-b92a-11e9-82c4-322f48ab2167.png)
Screenshot with Stacktrace

**Comment by markuslohn** [comment]
[javaee-customer-service.zip](https://github.com/crnk-project/crnk-framework/files/3477311/javaee-customer-service.zip)
Example for JAX-RS producing this error.

**Comment by markuslohn** [comment]
With the following fix in class JpaCriteriaQueryBackend I fixed the problem:

```
@Override
  public List<Order> getOrderList() {
    java.util.List<Order> orderedList = criteriaQuery.getOrderList();
    if (orderedList != null) {
      return new ArrayList<>(criteriaQuery.getOrderList());
    } else {
      return new ArrayList<>();
    }
  }
```

**Comment by remmeier** [comment]
never stumled upon it, but I applied your changes, thx!

**Comment by remmeier** [comment]
available as 3.1.x version on http://www.crnk.io/releases/latest/ shortly
